engine/src/main/java/org/terasology/rendering/dag/nodes/ToneMappingNode.java

```diff
@@ -63,16 +63,16 @@ public ToneMappingNode(Context context) {
         screenGrabber = context.get(ScreenGrabber.class);
 
         DisplayResolutionDependentFBOs displayResolutionDependentFBOs = context.get(DisplayResolutionDependentFBOs.class);
-        requiresFBO(new FBOConfig(TONE_MAPPING_FBO, FULL_SCALE, FBO.Type.HDR), displayResolutionDependentFBOs);
-        addDesiredStateChange(new BindFbo(TONE_MAPPING_FBO, displayResolutionDependentFBOs));
-        addDesiredStateChange(new SetViewportToSizeOf(TONE_MAPPING_FBO, displayResolutionDependentFBOs));
+        FBO toneMappingFbo = requiresFBO(new FBOConfig(TONE_MAPPING_FBO, FULL_SCALE, FBO.Type.HDR), displayResolutionDependentFBOs);
+        addDesiredStateChange(new BindFbo(toneMappingFbo));
+        addDesiredStateChange(new SetViewportToSizeOf(toneMappingFbo));
 
         addDesiredStateChange(new EnableMaterial(TONE_MAPPING_MATERIAL));
 
         toneMappingMaterial = getMaterial(TONE_MAPPING_MATERIAL);
 
         int textureSlot = 0;
-        addDesiredStateChange(new SetInputTextureFromFbo(textureSlot, INITIAL_POST_FBO, ColorTexture, displayResolutionDependentFBOs, TONE_MAPPING_MATERIAL, "texScene"));
+        addDesiredStateChange(new SetInputTextureFromFbo(textureSlot, displayResolutionDependentFBOs.get(INITIAL_POST_FBO), ColorTexture, displayResolutionDependentFBOs, TONE_MAPPING_MATERIAL, "texScene"));
```

Comment:
Actually, it all depends on how far we want to push the graph concept. 

One possible avenue is to have DataNodes (i.e. FBOs, cameras, perhaps shaders/materials) and RenderingNodes, the former being connected to the latter on graph construction. In this context it wouldn't be nodes that create everything but rather the objects that create the nodes. For example a module might inject its nodes into the DAG and also take care of creating all the connections between them and the existing DAG. 

I am happy to continue along the lines you mentioned though, leaving the concept above for the future, if ever.